src/messageComposer/attachmentManager.ts

```diff
@@ -453,8 +453,17 @@ export class AttachmentManager {
     if (localAttachment.localMetadata.uploadState === 'blocked') {
       this.upsertAttachments([localAttachment]);
       this.client.notifications.addError({
-        message: 'Error uploading attachment',
-        origin: { emitter: 'AttachmentManager', context: { attachment } },
+        message: `The attachment upload was blocked`,
+        origin: {
+          emitter: 'AttachmentManager',
+          context: { attachment, blockedAttachment: localAttachment },
+        },
+        options: {
+          code: 'attachment.upload.blocked',
+          metadata: {
+            reason: localAttachment.localMetadata.uploadPermissionCheck?.reason,
+          },
```

Comment:
Is it worth adding original error object at the same level the `reason` property is (if this is available in the UI)? `ErrorFromResponse` contains quite a bit of data which could be useful, like `code` and `status` numbers.

Reply:
But this not a error from a response. This is a reason calculated based on the back-end config.